src/subscribe/index.ts

```diff
@@ -29,28 +29,34 @@ export class Subscription extends Emitter {
     /** @internal */
     client: StreamrClient
     /** @internal */
-    options: Todo
+    options: ReturnType<typeof validateOptions> & {
+        id?: string
+    }
     /** @internal */
-    key: Todo
+    key
     /** @internal */
-    id: Todo
+    id
     /** @internal */
-    _onDone: Todo
+    _onDone: ReturnType<typeof Defer>
     /** @internal */
-    _onFinally: Todo
+    _onFinally
     /** @internal */
-    pipeline: Todo
+    pipeline
     /** @internal */
-    msgStream: Todo
```

Comment:
@teogeb FYI as of TypeScript 4.0, TS can can infer class members from usage in the constructor, so no need to list them all out with `Todo`/`any`, in most cases it can deduce the type exactly, without any annotation at all: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#class-property-inference-from-constructors.

Reply:
Yeah, true. Sometimes I put the types in anyway as it makes easier to read the code: you can just check the class variables and their types by reading the source code. But IDE hover hints help, if we skip that part 😊